File: TMessagesProj/src/main/java/ua/gomin/messenger/configs/GominAppearanceConfig.java
package ua.gomin.messenger.configs;

import android.content.Context;
import android.content.SharedPreferences;

public class GominAppearanceConfig {

    public static final GominAppearanceConfig INSTANCE = new GominAppearanceConfig();
    private static final String PREFS_NAME = "gomin_appearance_config";

    

    private SharedPreferences getPrefs(Context context) {
        return context.getSharedPreferences(PREFS_NAME, Context.MODE_PRIVATE);
    }

    public boolean getHideSearchFiled(Context context) {
        return getPrefs(context).getBoolean("hideSearchFiled", false);
    }
    public boolean getShowMainTabs(Context context) {
        return getPrefs(context).getBoolean("showMainTabs", false);
    }
    public boolean getProfileBackgroundColor(Context context) {
        return getPrefs(context).getBoolean("profileBackgroundColor", false);
    }
    public boolean getCenterTitle(Context context) {
        return getPrefs(context).getBoolean("centerTitle", false);
    }
    public boolean getDrawSnowInActionBar(Context context) {
        return getPrefs(context).getBoolean("drawSnowInActionBar", false);
    }
    public void setHideSearchFiled(Context context, boolean value) {
        getPrefs(context).edit().putBoolean("hideSearchFiled", value).apply();
    }
    public void setShowMainTabs(Context context, boolean value) {
        getPrefs(context).edit().putBoolean("showMainTabs", value).apply();
    }
    public void setProfileBackgroundColor(Context context, boolean value) {
        getPrefs(context).edit().putBoolean("profileBackgroundColor", value).apply();
    }
    public void setCenterTitle(Context context, boolean value) {
        getPrefs(context).edit().putBoolean("centerTitle", value).apply();
    }
    public void setDrawSnowInActionBar(Context context, boolean value) {
        getPrefs(context).edit().putBoolean("drawSnowInActionBar", value).apply();
    }
}

File: TMessagesProj/src/main/java/ua/gomin/messenger/configs/GominAppearanceConfig.kt
package ua.gomin.messenger.configs

import android.content.Context
import android.content.SharedPreferences

/**
 * Налаштування зовнішнього вигляду.
 */
object GominAppearanceConfig {

    private const val PREFS_NAME = "gomin_appearance_config"

    private fun getPrefs(context: Context): SharedPreferences {
        return context.getSharedPreferences(PREFS_NAME, Context.MODE_PRIVATE)
    }

    fun getHideSearchFiled(context: Context): Boolean =
        getPrefs(context).getBoolean("hideSearchFiled", false)

    fun setHideSearchFiled(context: Context, hidden: Boolean) {
        getPrefs(context).edit().putBoolean("hideSearchFiled", hidden).apply()
    }

    fun getShowMainTabs(context: Context): Boolean =
        getPrefs(context).getBoolean("showMainTabs", false)

    fun setShowMainTabs(context: Context, enabled: Boolean) {
        getPrefs(context).edit().putBoolean("showMainTabs", enabled).apply()
    }

    fun getProfileBackgroundColor(context: Context): Boolean =
        getPrefs(context).getBoolean("profileBackgroundColor", false)

    fun setProfileBackgroundColor(context: Context, enabled: Boolean) {
        getPrefs(context).edit().putBoolean("profileBackgroundColor", enabled).apply()
    }

    fun getCenterTitle(context: Context): Boolean =
        getPrefs(context).getBoolean("centerTitle", false)

    fun setCenterTitle(context: Context, enabled: Boolean) {
        getPrefs(context).edit().putBoolean("centerTitle", enabled).apply()
    }

    fun getDrawSnowInActionBar(context: Context): Boolean =
        getPrefs(context).getBoolean("drawSnowInActionBar", false)

    fun setDrawSnowInActionBar(context: Context, enabled: Boolean) {
        getPrefs(context).edit().putBoolean("drawSnowInActionBar", enabled).apply()
    }
}
